### PR TITLE
inline Deposits & Disputes tracks event names & remove constants from tracks/index.js

### PR DESCRIPTION
**Pull request by haszari**
Part of #8066

Note this PR only fixes Helix-owned events, so that issue should stay open when this PR is merged.

#### Changes proposed in this Pull Request

This PR inlines all deposits and disputes related tracks event names, and removes the now-redundant constants from the tracks/index file.

The purpose of this is to make it easier to work on and inspect tracks events by searching the code base, and remove the constants which are an unnecessary layer of indirection. See #8066 for full rationale.

**Bonus bug fix** Also in this PR I've made sure that `wcpay_disputes_row_action_click` fires no matter where you click in the row. The cells/row link through to transaction details, exact same action as the `Respond` button, but (previously) the event only fired for the button. See 7a61e3e4af075d08a9563dbf2be9c8c166b4a57f 

#### Testing instructions

TLDR: the events should all fire as before with no changes. 

##### Get set up to trigger and view tracks events
- Ensure your site is opted in to usage tracking. `Dashboard > WooCommerce > Settings > Advanced > WooCommerce.com`, check `Allow usage of WooCommerce to be tracked`, save settings.
- Open browser dev tools, click `Network` tab.
- Search for `t.gif` to filter network requests.
- Each `GET` request for `t.gif` is a tracks event. 
- View `Payload` to see event name `_en` and other props.
- Read on for testing tips for each event…

There are other ways to view events, feel free to use whatever approach you like.

##### Test flows with events

Test some/all of the following events fire correctly:

- Event name (`_en`) is correct.
- Event props are present and have meaningful values.

To determine correct props, you can compare with prod or old build, look at tracks in tracks backend, or inspect the code.

###### Deposits
- `wcadmin_wcpay_deposits_row_click` Go `Payments > Deposits` and click a deposit to view details.
- `wcadmin_wcpay_deposits_download` Go `Payments > Deposits` and click `Download` to export deposits as CSV.
- `wcadmin_wcpay_overview_deposits_view_history_click` Go `Payments > Overview`, scroll to `Deposits` box, click `Show full deposits history`.
- `wcadmin_wcpay_overview_deposits_change_schedule_click` Go `Payments > Overview`, scroll to `Deposits` box, click `Change deposit schedule`.
- `wcadmin_wcpay_settings_deposits_manage_in_stripe_click` Go `Payments > Settings`, scroll to `Deposits`, click `Manage in Stripe`.

###### Disputes
To test these you'll need disputes in a variety of states – open / needs response, inquiry / dispute proper, etc. Refer to … our internal docs for how to set these states up (we should document all these flows in repo wiki!).

- `wcadmin_wcpay_disputes_row_action_click` Go `Payments > Disputes` and click the button (rightmost column) for a dispute – it will be `Respond` or `See details` depending on status.
  - ~~Discovered a bug with this. If we click the button, we get the event. If we click the row, the same exact action happens (view details) but there's no event.~~
  - [x] TBD, might fix this in this PR or add a follow up issue 
    - Fixed! 7a61e3e4af075d08a9563dbf2be9c8c166b4a57f
- `wcadmin_wcpay_dispute_challenge_clicked` Go `Payments > Disputes`, and click a live dispute or inquiry. In transaction detail screen, click `Challenge dispute` button.
- `wcadmin_wcpay_view_submitted_evidence_clicked` Go `Payments > Disputes`, and click a dispute or inquiry that needs response. Fill out the challenge form and submit. Go back to the transaction detail – view `All disputes` on list view then click the relevant row. Click `View submitted evidence`.
- Accepting a dispute:
  - `wcadmin_wcpay_dispute_accept_modal_view` Go `Payments > Disputes`, and click a live dispute. In transaction detail screen, click `Accept dispute` button.
  - `wcadmin_wcpay_dispute_accept_click` Follow steps for `wcadmin_wcpay_dispute_accept_modal_view` then click `Accept dispute` in the modal.
- Refunding a pre-dispute inquiry:
  - Buy something with `4000000000001976` card (simulates an inquiry).
  - `wcadmin_wcpay_dispute_inquiry_refund_modal_view` Go `Payments > Disputes`, and click the inquiry. Click `Issue refund` button.
  - `wcadmin_wcpay_dispute_inquiry_refund_click` Follow steps for `wcadmin_wcpay_dispute_inquiry_refund_modal_view` then click `View order to issue refund` button
    - [x] This event name is misleading, `wcadmin_wcpay_dispute_inquiry_refund_view_order` might be better.
    - Ignoring this for now, the docs for this event are clear that it navigates to order. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️) n/a - manual testing is reasonable here, no merchant facing changes
- [x] Tested on mobile (or does not apply) n/a - no UI changes

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8069` or branch name `tidy/remove-helix-tracks-names-constants` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=tidy/remove-helix-tracks-names-constants) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9100fca0c9c30c91b8f954f60208d384129cadb7
- Build time: 2024-02-02 02:48:51 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/disputes/index.tsx [hunk 1]
```diff
@@ -205,9 +205,20 @@ export const DisputesList = (): JSX.Element => {
...
+			e.preventDefault();
+			wcpayTracks.recordEvent( 'wcpay_disputes_row_action_click' );
+			const history = getHistory();
+			history.push( getDetailsURL( dispute.charge_id, 'transactions' ) );
+		};
+
 		const clickable = ( children: React.ReactNode ): JSX.Element => (
 			<ClickableCell
 				href={ getDetailsURL( dispute.charge_id, 'transactions' ) }
+				onClick={ onClickDisputeRow }
```
Comment (Jinksi) [inline client/disputes/index.tsx:221]: 💭 The event name `wcpay_disputes_row_action_click` and description point towards this event being specific to the action button rather than the row.

```
This event measures when the action button on a dispute row is clicked.
```

Should the user action of clicking the row be recorded as separate event? i.e. `wcpay_disputes_row_click`. This would be more consistent with deposits: `wcpay_deposits_row_click`.

Or do you think it's OK for us to apply this event name loosely, since it shares the same user intent (viewing the details of a dispute)?

File: changelog/tidy-remove-helix-tracks-names-constants [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Refactor dispute & deposits tracks event names inline. `wcpay_disputes_row_action_click` now also fires from row/cell clicks, not just the action button.
```
Comment (Jinksi) [inline changelog/tidy-remove-helix-tracks-names-constants:4]: ⛏️ I would consider making this a "Comment:" placeholder changelog entry, since it has no user-facing changes.

Or, at least, the `Refactor dispute & deposits tracks event names inline. ` part is not user-facing.

**Review by Jinksi — APPROVED** [review]
Nice one @haszari 🌟

I've checked the payload fired from various dispute/deposit UI interactions and had a close inspection of the code changes to ensure the event names have been moved inline accurately. ✅

**Comment by Jinksi** [comment]
@haszari Note that this approach of inlining the event names conflicts with the approach of the recently merged PR https://github.com/Automattic/woocommerce-payments/pull/8106/files#diff-60fec635c41f548c0b7ab9632831c145966c4bae9619a6d7ed94c1c9302500cd which creates an `events.ts` file to hold all of the tracks event names.

**Review by Jinksi — COMMENTED** [review]
@haszari this PR's approach of inlining the event names conflicts with the approach of the recently merged PR #8106 which has moved all event names into an `events.ts` file.

I don't think we can merge this whilst knowingly going against the proposed standardised `events.ts` approach without further discussion.

File: client/disputes/index.tsx (hunk 1 above)
Comment (haszari) [reply client/disputes/index.tsx:221]: > Or do you think it's OK for us to apply this event name loosely, since it shares the same user intent (viewing the details of a dispute)?

I think the button and the row represent the same user action, so am intentionally proposing same event for all. I don't think we want to micro-optimise our UI like that (yet!), if we want to understand merchant behaviour when viewing and actioning disputes, we want to know when the click through to detail view. 

Good point about the event docs, I should update that as a follow up. 

> Should the user action of clicking the row be recorded as separate event? i.e. wcpay_disputes_row_click. This would be more consistent with deposits: wcpay_deposits_row_click.

That's a good comparison. In that table we have a hyperlinked date, and we don't track that separately from clicking the row.

If we were curious to optimise or understand micro details like this, we might add these events … but IMO we need coarse events covering the main actions merchants can take. E.g. view list, click to see detail, etc.

**Comment by Jinksi** [comment]
@haszari Should we remove this from the [7.2 milestone](https://github.com/Automattic/woocommerce-payments/milestone/39) (Feb 11)?

**Comment by haszari** [comment]
> Should we remove this from the [7.2 milestone](https://github.com/Automattic/woocommerce-payments/milestone/39) (Feb 11)?

Good idea, it's getting close and there's no urgency here – now removed 👍🏻

**Comment by haszari** [comment]
Closing - now redundant, superceded by:

- #8166
